src/Product/ProductInfo.cs

```diff
@@ -8,44 +8,42 @@ namespace Azure.DataApiBuilder.Product;
 
 public static class ProductInfo
 {
-    public const string DEFAULT_VERSION = "0.0.0";
     public const string DAB_APP_NAME_ENV = "DAB_APP_NAME_ENV";
-    public static readonly string DEFAULT_APP_NAME = $"dab_oss_{ProductInfo.GetProductVersion()}";
-    public static readonly string ROLE_NAME = "DataApiBuilder";
+    public static readonly string DAB_USER_AGENT = $"dab_oss_{GetProductVersion()}";
+    public static readonly string CLOUD_ROLE_NAME = "DataApiBuilder";
 
     /// <summary>
-    /// Reads the product version from the executing assembly's file version information.
+    /// Returns the Product version in Major.Minor.Patch format without a commit hash.
+    /// FileVersionInfo.ProductBuildPart is used to represent the Patch version.
+    /// FileVersionInfo is used to retrieve the version information from the executing assembly
+    /// set by the Version property in Directory.Build.props
     /// </summary>
-    /// <returns>Product version if not null, default version 0.0.0 otherwise.</returns>
-    public static string GetProductVersion()
+    /// <param name="includeCommitHash">If true, returns the version string with the commit hash</param>
+    /// <returns>Version string "Major.Minor.Patch"</returns>
+    public static string GetProductVersion(bool includeCommitHash = false)
     {
         Assembly assembly = Assembly.GetExecutingAssembly();
         FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(assembly.Location);
-        string? version = fileVersionInfo.ProductVersion;
 
-        return version ?? DEFAULT_VERSION;
+        if (includeCommitHash)
```

Comment:
#1456 can be used to track the issue with `dab --version` alone. We can still prevent the hash from occurring in the first line of individual commands like `dab update` if we dont add this `includeCommitHash` parameter altogether.

Reply:
Thanks for pointing out. I too eagerly reverted to using commit hash. Commit hash will only be printed when: CLI usage with `--help`, `--version` , or unknown verbs/options are used.
Tests and CLI options parsing now properly only prints version without commit hash where possible.